### PR TITLE
New Brunswick Age Amount Credit

### LINKED ISSUE #271: New Brunswick Age amount credit
[Reference](https://www.unb.ca/hr/_assets/documents/td1nb-e.pdf)

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #271

### CODE REVIEW

File: policyengine_canada/parameters/gov/provinces/nb/tax/income/credits/age_amount/age_eligibility.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+description: New Brunswick allows people to access the age amount tax credit past this age.
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/nb/tax/income/credits/age_amount/age_eligibility.yaml:1]: > vs >= was ambiguous
```suggestion
description: New Brunswick provides an age amount tax credit to people this age or older.
```

File: policyengine_canada/parameters/gov/provinces/nb/tax/income/credits/age_amount/age_eligibility.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+description: New Brunswick allows people to access the age amount tax credit past this age. 
+values:
+  2023-01-01: 65
+metadata:
+  unit: year
+  label: New Brunswick age amount credit age eligibility
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/nb/tax/income/credits/age_amount/age_eligibility.yaml:6]:
```suggestion
  label: New Brunswick age amount credit age minimum
```

File: policyengine_canada/parameters/gov/provinces/nb/tax/income/credits/age_amount/reduction_rate.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+description: New Brunswick reduces the age amount tax credit at this rate.
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-CAD
+  rate_unit: /1
+  label: New Brunswick age amount credit reduction rate.
+  reference:
+  - title: 2023 New Brunswick Personal Tax Credit Return
+    href: https://www.unb.ca/hr/_assets/documents/td1nb-e.pdf
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/nb/tax/income/credits/age_amount/reduction_rate.yaml:9]: This seems like the better source, specifies 15% directly: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1nb-ws/td1nb-ws-23e.pdf

File: policyengine_canada/variables/gov/provinces/nb/tax/income/credits/age_amount/nb_age_amount_credit.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    entity = Person
+    label = "New Brunswick age amount credit"
+    definition_period = YEAR
+    defined_for = ProvinceCode.NB
+
+    def formula(person, period, parameters):
+        age = person("age", period)
+        income = person("individual_net_income", period)
+        p = parameters(period).gov.provinces.nb.tax.income.credits.age_amount
+        eligibility = age >= p.age_eligibility
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nb/tax/income/credits/age_amount/nb_age_amount_credit.py:15]: I think this is our convention
```suggestion
        eligible = age >= p.age_eligibility
```

File: policyengine_canada/variables/gov/provinces/nb/tax/income/credits/age_amount/nb_age_amount_credit.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        reduced_amount = p.amount - p.reduction_rate.calc(income)
+        return eligibility * (max_(0, reduced_amount))
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nb/tax/income/credits/age_amount/nb_age_amount_credit.py:17]:
```suggestion
        return eligibility * max_(0, reduced_amount)
```

**Review by MaxGhenis — APPROVED** [review]
Good to merge after these suggestions

**Comment by PavelMakarchuk** [comment]
> Good to merge after these suggestions

done

**Pull request merged**